spec/Naming.md

```diff
@@ -0,0 +1,120 @@
+# Naming
+
+We define the unique name strategy per datasource to ensure it is followed uniformly independently from the type of job consuming and producing data.
+The namespace for a dataset is the unique name for its datasource.
+For jobs it is the scheduler.
+
+## Datasets:
+The namespace and name of a datasource can be combined to form a URI (scheme:[//authority]path)
+* Namespace = scheme:[//authority] (the datasource)
+* Name = path (the datasets)
+
+### Data warehouses/data bases.
+Datasets are called tables. Tables are organised in databases and schemas.
+#### Postgres:
+Datasource hierarchy:
+ * Host
+ * Port
+
+Naming hierarchy:
+ * Database
+ * Schema
+ * Table
+
+Identifier:
+ * Namespace: postgres://{host}:{port} of the service instance.
+   * Scheme = postgres
+   * Authority = {host}:{port}
+ * Unique name: {database}.{schema}.{table}
+   * URI =  postgres://{host}:{port}/{database}.{schema}.{table}
+  
+#### Snowflake
+See: [Object Identifiers — Snowflake Documentation](https://docs.snowflake.com/en/sql-reference/identifiers.html)
+
+Datasource hierarchy:
+ * account name
+
+Naming hierarchy:
+ * Database: {database name} => unique across the account
+ * Schema: {schema name} => unique within the database
+ * Table: {table name} => unique within the schema
+
+Identifier:
+ * Namespace: snowflake://{account name}
+   * Scheme = snowflake
+   * Authority = {account name}
+ * Name: {database}.{schema}.{table}
+   * URI =   snowflake://{account name}/{database}.{schema}.{table}
+
+#### BigQuery
+See: 
+[Creating and managing projects | Resource Manager Documentation](https://cloud.google.com/resource-manager/docs/creating-managing-projects)
+[Introduction to datasets | BigQuery](https://cloud.google.com/bigquery/docs/datasets-intro)
+[Introduction to tables | BigQuery](https://cloud.google.com/bigquery/docs/tables-intro)
+
+Datasource hierarchy:
+ * bigquery
+
+Naming hierarchy:
+ * Project Name: {project name} => is not unique
+ * Project number: {project number} => numeric: is unique across google cloud
+ * Project ID: {project id} => readable: is unique across google cloud
+ * dataset: {dataset name} => is unique within a project
+ * table: {table name} => is unique within a dataset
+
+Identifier :
+ * Namespace: bigquery
+   * Scheme = bigquery
+   * Authority = 
+ * Unique name: {project id}.{dataset name}.{table name}
+   * URI =   bigquery:{project id}.{schema}.{table}
+
+### Distributed file systems/blob stores
+#### GCS
+Datasource hierarchy: none, naming is global
+
+Naming hierarchy:
+ * bucket name => globally unique
+ * Path 
+
+Identifier :
+ * Namespace: gs://{bucket name}
+   * Scheme = gs
+   * Authority = {bucket name}
+ * Unique name: {path}
+   * URI =   gs://{bucket name}{path}
+
+#### S3
+Naming hierarchy:
+ * bucket name => globally unique
+ * Path 
+
+Identifier :
+ * Namespace: s3://{bucket name}
```

Comment:
A quick question here surrounding this namespace. I know the suggestion is to use the s3 path but looking at the Open Lineage documentation it says that Namespaces should only have letters and numbers. Does this mean there will be an additional change that will allow '/' to be included in a namespace? Thank you

Reply:
Yes. This has been updated in Marquez ":" and "/" are now allowed. can you point me to the doc you are looking at, we need to update this?

Reply:
Hey Julien, thank you for getting back to me here, I'm assuming this change will be in 0.14.0? The documentation I was looking at was here in the API reference documentation: https://marquezproject.github.io/marquez/openapi.html#tag/Namespaces/paths/~1namespaces~1{namespace}/put

Reply:
It looks like the documentation was not updated when that change was made in 0.13:
https://github.com/MarquezProject/marquez/commit/4e2cee04e2040bf55d37a2a29c0c981cda016717
We're going to fix it.
It is in 0.13.1 already.

Reply:
Ah ok. Looks like we behind on versions anyways. I'll look into updating to 0.13.1. I believe my colleague @KevinMellott91 contributed to Marquez by making that Java 8 compatible, which was our original issue with updating. Thank you for looking into this Julien!